### PR TITLE
Document best practices for setting up a good development environment.

### PR DESCRIPTION
**Pull request by jacob314**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,3 +30,27 @@ From the devtools directory:
...
+## Deployment
+
+The strategy to deploy these performance tools has not yet been finalized.
+
+One aspect to be aware of is that the development steps build the
+application using [dartdevc](https://webdev.dartlang.org/tools/dartdevc)
+which provides a productive development experience while the application will
+be deployed using the Dart2Js compiler which generates significantly faster
+JS. Before optimizing slow code, make sure there is still a performance issue
+when running the application with dart2js.
```
Comment (kenzieschmoll) [inline README.md]: I'd suggest rewording this paragraph so that it is less of a run-on sentence. It was a little hard to parse on first glance. This would be my crack at it:

The development steps build the application using [dartdevc](https://webdev.dartlang.org/tools/dartdevc), which provides a productive development experience. The application, however, will be deployed using the Dart2Js compiler, which generates significantly faster JS. Before optimizing slow code, make sure there is still a performance issue when running the application with Dart2js.

Comment (kenzieschmoll) [inline README.md]: Capitalize dart2js for consistency.

File: README.md [hunk 2]
```diff
@@ -30,3 +30,27 @@ From the devtools directory:
...
+For more productive development, launch your flutter application specifying
+`--observatory-port` so the observatory is available on a fixed port. This
+lets you avoid manually entering the observatory port paremeter each time
+you launch the application.
+
+- flutter run --observatory-port=8888
+- open http://localhost:8080/?port=8888
+
+`pub run webdev` provides an fast development server that incrementally
+rebuilds portion of the application that was edited each time you reload the
```
Comment (kenzieschmoll) [inline README.md]: an -> a

Comment (kenzieschmoll) [inline README.md]: -> ...rebuilds **the** portion...

File: README.md (hunk 1 above)
Comment (jacob314) [reply README.md]: done

File: README.md (hunk 2 above)
Comment (jacob314) [reply README.md]: done

Comment (jacob314) [reply README.md]: done

File: README.md [hunk 3]
```diff
@@ -30,3 +30,27 @@ From the devtools directory:
 Then, open a browser window to the local url specified by webdev. After the page has loaded, append
 `?port=xxx` to the url, where xxx is the port number of the service protocol port, as specified by 
 the `flutter run` output.
+
+For more productive development, launch your flutter application specifying
+`--observatory-port` so the observatory is available on a fixed port. This
+lets you avoid manually entering the observatory port paremeter each time
```
Comment (devoncarew) [inline README.md]: sp: parameter
